### ISSUE TITLE
this.reverse should work OOTB

### ISSUE THREAD

**Issue by pzuraq** [pzuraq/liquid-tether#23]
Currently a tether needs to observe its context in order to know what animation it should use with something like `toValue` and `fromValue`. Ideally, for the simple case of a modal that just enters and exits the screen, `reverse` should work as one would expect.

**Comment by knownasilya** [comment]
My workaround looks like this, for anyone interested:

```js
this.transition(
    target('disclaimer-modal'),
    // hack to get reverse working..
    this.toValue(current => !!current.index),
    this.use('tether', ['to-down', options]),
    this.reverse('tether', ['to-up', options]),
  );
```

**Comment by richardvenneman** [comment]
@knownasilya Is this the complete workaround or am I missing something on the component level? Transitions stop working completely when specifying the `toValue`:

```javascript
this.transition(
  target('booking-details'),
  
  this.toValue(current => !!current.index),

  this.use('tether',
    ['to-up', { duration: 500, easing: 'easeInOutQuint' }],
    ['fade', { duration: 200 }]),

  this.reverse('tether',
    ['fade', { duration: 200 }],
    ['fade', { duration: 300 }])
);
```

**Comment by knownasilya** [comment]
Here's a working example https://github.com/knownasilya/bootstrap-modal/blob/master/addon/utils/default-transition.js#L13

**Comment by pzuraq** [comment]
@richardvenneman you may have to do something like:

```
this.toValue((current) => !!current.index)
```

Not sure if the above is completely valid ES6. That should work though, `index` should be increased for every tether added to a target starting at 0, so from 1 -> current.index of the `toValue` would be truthy and current.index of the `fromValue` would be falsy.

**Comment by richardvenneman** [comment]
@pzuraq unfortunately that did not work (also tried that as per my previous comment).

@knownasilya's trick works though! Thanks a ton.

```javascript
this.toValue(current => {
  return current.isVisible;
}),
```

**Comment by pzuraq** [comment]
I just made #36 which adds two new helpers which should make this a bit easier. There is `openTether` which will cause the `use` animation to trigger for all animations on a single target except for when the tether is transitioning out entirely (to an empty state). In that case, the `reverse` transition will be used. A symmetric `closeTether` constraint has also been added.

Would love your thoughts on this @knownasilya and @richardvenneman, seems like a simple way to solve the problem without losing the ability to add more context for more complicated modal transitions.
